### PR TITLE
[#91] Move metadata to external contract

### LINKED ISSUE #91: Move metadata into another contract
# Clarification and motivation

In preparation for TZIP-16 (that's currently WIP) we should extract metadata outside of our contract. So `token_metadata_registry` shouldn't return `SELF` anymore.

# Acceptance criteria

1. The `stablecoin` contract doesn't store metadata anymore. It stores an address of a contract that holds metadata instead.
2. TBD: should it be possible to update that address?
3. The repository has a simple metadata contract that only holds metadata in its storage and doesn't do anything else. I suppose we can implement it in vanilla Michelson because it's really trivial (I suppose it can always `FAILWITH` for example).
4. Tests are updated to use that metadata contract.

### PR DESCRIPTION
**Pull request by sras**
## Related issue(s)

Resolves #91 

#### Related changes (conditional)

- Tests
  - [ ] If I added new functionality, I added tests covering it.
  - [ ] If I fixed a bug, I added a regression test to prevent the bug from
        silently reappearing again.

- Documentation
  - [ ] I checked whether I should update the docs and did so if necessary:
    - [README](../tree/master/README.md)
    - Haddock
  - [ ] I updated [changelog](../tree/master/ChangeLog.md) unless I am sure my changes are
        not essential.

#### Stylistic guide (mandatory)

- [ ] My commits comply with [the policy used in Serokell](https://www.notion.so/serokell/Where-and-how-to-commit-your-work-58f8973a4b3142c8abbd2e6fd5b3a08e).
- [ ] My code complies with the [style guide](../tree/master/docs/code-style.md).

### CODE REVIEW

File: haskell/test/resources/metadata.tz [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{ parameter unit ;
+  storage
+    (pair (unit %dummy_field)
+          (big_map %token_metadata
+             nat
+             (pair (nat %token_id)
+                   (pair (string %symbol)
+                         (pair (string %name) (pair (nat %decimals) (map %extras string string))))))) ;
+  code { CDR ; NIL operation ; PAIR } }
+
```
[the remark this answers is not in the document]
Comment (gromakovsky) [reply haskell/test/resources/metadata.tz:10]: @sras I'm ok with that, but then I think we also need a CI check that `ligo compile-contract metadata.ligo` is the same as `test/resources/metadata.tz`. And we need to check that `ligo compile-contract metadata.ligo` works anyway. I believe it should be straightforward to add this check.

**Comment by sras** [comment]
>Did you test that both origination scenarios work in client? 1. Manually deployed metadata contract passed from CLI. 2. No contract is passed, it's automatically originated.

Yes. This appear to work.

**Comment by sras** [comment]
I have removed the `metadata.tz` from repo and have changed the pipeline. I have not merged it since I am not sure about the changed made to the nix files, and think it would be better to merge after it is reviewed.

File: .reuse/dep5 [hunk 2]
```diff
@@ -1,9 +1,9 @@
 Format: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/
 
-Files: .github/* nix/sources.json
+Files: .github/* nix/sources.json haskell/test/resources/metadata.tz
```
Comment (gromakovsky) [inline .reuse/dep5:3]: This file should be reverted, I guess.

**Comment by gromakovsky** [comment]
LGTM

File: .reuse/dep5 (hunk 2 above)
Comment (sras) [reply .reuse/dep5]: Reverted

**Pull request merged**